crates/biome_js_formatter/src/js/expressions/arrow_function_expression.rs

```diff
@@ -494,7 +500,23 @@ impl Format<JsFormatContext> for ArrowChain {
                     ])]
                 )?;
             } else {
-                write!(f, [format_tail_body])?;
+                let should_add_parens = should_add_parens(&tail_body);
+                write!(
+                    f,
+                    [format_with(|f| {
+                        if should_add_parens {
+                            write!(f, [if_group_fits_on_line(&text("("))])?;
+                        }
+
+                        write!(f, [format_tail_body])?;
+
+                        if should_add_parens {
+                            write!(f, [if_group_fits_on_line(&text(")"))])?;
+                        }
```

Comment:
It's your choice🙂
```suggestion
                        if should_add_parens {
                            write!(
                                f,
                                [
                                    if_group_fits_on_line(&text("(")),
                                    format_body,
                                    if_group_fits_on_line(&text(")"))
                                ]
                            )?;
                        } else {
                            write!(f, [format_body])?;
                        }
```

Reply:
Thanks! I've remove duplicated if block.